blogging/views.py

```diff
@@ -0,0 +1,144 @@
+from django.shortcuts import render, redirect
+from django.http import HttpResponse, HttpResponseRedirect, Http404
+from django.template import loader
+from django.utils import timezone
+from blogging.models import Post, Category
+from blogging.forms import PostForm, PublishForm, ListForm
+
+
+def list_view(request):
+    """
+    Displays a list view of each published post.
+
+    In the real world I'd likely want to limit the number of posts displayed incase
+    the blog grows large, with a 'display more' at the bottom.
+    Don't know how to do that yet though...
+    """
+    if request.method == "POST":
+        if not request.user.is_authenticated:
+            return redirect("/login")
```

Comment:
you use this a few times here, you could likely make it a function for authentication redirect.